public/app/plugins/panel/dashlist/DashList.tsx

```diff
@@ -0,0 +1,142 @@
+
+      let recentDashboards: Promise<Dashboard[]> = Promise.resolve([]);
+      if (props.options.showRecentlyViewed) {
+        const dashIds = take(impressionSrv.getDashboardOpened(), props.options.maxItems);
+        recentDashboards = getBackendSrv().search({ dashboardIds: dashIds, limit: props.options.maxItems });
+      }
+
+      let searchedDashboards: Promise<Dashboard[]> = Promise.resolve([]);
+      if (props.options.showSearch) {
+        const params = {
+          limit: props.options.maxItems,
+          query: props.options.query,
+          folderIds: props.options.folderId,
+          tag: props.options.tags,
+          type: 'dash-db',
+        };
+
+        searchedDashboards = getBackendSrv().search(params);
+      }
+
+      const [starred, searched, recent] = await Promise.all([starredDashboards, searchedDashboards, recentDashboards]);
+      const dashMap = starred.reduce(
+        (acc, dash) => Object.assign(acc, { [dash.id]: dash }),
+        {} as Record<number, Dashboard>
+      );
+
+      searched.forEach(dash => {
+        if (dashMap.hasOwnProperty(dash.id)) {
+          dashMap[dash.id].isSearchResult = true;
+        } else {
+          dashMap[dash.id] = { ...dash, isSearchResult: true };
+        }
+      });
+
+      recent.forEach(dash => {
+        if (dashMap.hasOwnProperty(dash.id)) {
+          dashMap[dash.id].isRecent = true;
+        } else {
+          dashMap[dash.id] = { ...dash, isRecent: true };
+        }
+      });
+
+      setDashboards(dashMap);
+    }
+
+    fetchDashboards();
+  }, [
+    props.options.showSearch,
+    props.options.showStarred,
+    props.options.showRecentlyViewed,
+    props.options.maxItems,
+    props.options.query,
+    props.options.tags,
+    props.options.folderId,
+  ]);
+
+  const toggleDashboardStar = async (e: React.SyntheticEvent, dash: Dashboard) => {
+    e.preventDefault();
+    e.stopPropagation();
+
+    const isStarred = await getDashboardSrv().starDashboard(dash.id.toString(), dash.isStarred);
+    setDashboards(Object.assign({}, dashboards, { [dash.id]: { ...dash, isStarred } }));
+  };
+
+  const { showStarred, showRecentlyViewed, showHeadings, showSearch } = props.options;
+
+  const dashboardList = Object.values(dashboards);
+  const dashboardGroups: DashboardGroup[] = [
+    {
+      header: 'Starred dashboards',
+      dashboards: dashboardList.filter(dash => dash.isStarred),
+      show: showStarred,
+    },
+    {
+      header: 'Recently viewed dashboards',
+      dashboards: dashboardList.filter(dash => dash.isRecent),
+      show: showRecentlyViewed,
+    },
+    {
+      header: 'Search',
+      dashboards: dashboardList.filter(dash => dash.isSearchResult),
+      show: showSearch,
+    },
+  ];
+
+  return (
+    <CustomScrollbar autoHeightMin="100%" autoHeightMax="100%">
+      {dashboardGroups.map(
+        ({ show, header, dashboards }, i) =>
+          show && (
+            <div className="dashlist-section" key={`dash-group-${i}`}>
```

Comment:
Not sure if it's a part of the task, but would be nice to move the styles from these classes to emotion.

Reply:
Agree with @Clarity-89. Would be great if we could migrate these styles to Emotion